substrate/frame/staking-async/src/tests/unbonding_queue.rs

```diff
@@ -0,0 +1,603 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+use crate::{
+	pallet::normalize_era, session_rotation::EraElectionPlanner, tests::Test, UnbondingQueueConfig,
+	UnbondingQueueParams,
+};
+use sp_npos_elections::Support;
+use sp_runtime::{traits::Zero, Perbill};
+
+#[test]
+fn get_min_lowest_stake_works() {
+	ExtBuilder::default()
+		.set_status(31, StakerStatus::<AccountId>::Validator)
+		.validator_count(3)
+		.build_and_execute(|| {
+			assert_ok!(Staking::set_staking_configs(
+				RuntimeOrigin::root(),
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Set(UnbondingQueueConfig {
+					min_slashable_share: Perbill::from_percent(50),
+					lowest_ratio: Perbill::from_percent(34),
+					unbond_period_lower_bound: 1,
+					back_of_unbonding_queue: Zero::zero(),
+				})
+			));
+
+			// Check the era we are working with.
+			assert_eq!(Staking::get_min_lowest_stake(), 0);
+			Session::roll_until_active_era(2);
+			let current_era = Staking::current_era();
+			assert_eq!(current_era, 2);
+			assert_eq!(Staking::get_min_lowest_stake(), 500);
+
+			// Start initial era and verify setup.
+			assert_eq!(
+				UnbondingQueueParams::<Test>::get().unwrap(),
+				UnbondingQueueConfig {
+					min_slashable_share: Perbill::from_percent(50),
+					lowest_ratio: Perbill::from_percent(34),
+					unbond_period_lower_bound: 1,
+					back_of_unbonding_queue: Zero::zero(),
+				}
+			);
+
+			// Populate some `EraLowestRatioTotalStake` entries to test the function.
+			let bonding_duration = <Test as Config>::BondingDuration::get();
+			for i in current_era + 1..bonding_duration {
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(11), 100));
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(21), 100));
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(31), 100));
+				assert_eq!(Staking::get_min_lowest_stake(), 500);
+				Session::roll_until_active_era(i + 1);
```

Comment:
You are already in era 2, this will be a noop in the first round.

Reply:
Completely refactored in [aa3d40c](https://github.com/paritytech/polkadot-sdk/pull/8298/commits/aa3d40c2cb608696e9669c2a5f1bea022fd4848d).